# Basic Descriptives

### This script contains the following points:

#### 01. Importing libraries
#### 02. Importing orders.csv
#### 03. Checking orders.csv data column headings
#### 04. Importing products.csv data
#### 05. Examining products.csv data
#### 06. Examining orders.csv data

## 01. Importing libraries

In [1]:
# Import libraries
import pandas as pd
import numpy as np
import os

## 02. Importing orders.csv data

In [2]:
# Assign the main project folder path to the variable path
path = r'/Users/elsaekevall/Jupyter_Notebook/Career_Foundry/06_2022_Instacart_Basket_Analysis/'

In [3]:
# display the path
path

'/Users/elsaekevall/Jupyter_Notebook/Career_Foundry/06_2022_Instacart_Basket_Analysis/'

In [4]:
# Create a variable list of the headings to be imported
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [5]:
# display the vars_list
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [6]:
# Use the os.path.join() function to import the orders.csv file as pandas a dataframe with only the varaibles listed in vars_list
df = pd.read_csv(os.path.join(path, '02_Data', '02_1_Original_Data', 'orders.csv'), usecols = vars_list)

## 03. Checking orders.csv data column headings

*The directions in the exercise say "Print the first 5 rows of your dataframe" which I find very strange, but have done it here anyway. Then just below I have used the df.head() function to view the data and subsequently for the products dataframe have used only this function and the tails function wihtout adding print. To me at this stage it makes more sense to only view the data.*

In [7]:
# print the first five rows of the orders dataframe
print(df.head())

   order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0  


In [8]:
# Use the df.head function to view the first five rows of the dataframe
df.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

## 04. Importing products.csv data

In [9]:
# Use the os.path.join() function to import the products.csv file as pandas a dataframe without the index column
df_prods = pd.read_csv(os.path.join(path, '02_Data', '02_1_Original_Data', 'products.csv'), index_col = False)

## 05. Examining products.csv data

In [10]:
# Use the df.head function to view the first twenty rows of the products dataframe
df_prods.head(20)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

In [11]:
# Use the df.tail function to view the last thirty five rows of the products dataframe
df_prods.tail(35)

product_id                                       product_name  \
49658       49654                   Teriyaki Sauce, Sesame, Original   
49659       49655                                        Apple Cider   
49660       49656                         Masada Kosher Pocket Bread   
49661       49657                                  Cabernet Tomatoes   
49662       49658                         Brie with Herbs Foil Wedge   
49663       49659                            Organic Creamed Coconut   
49664       49660                        Professionals Sleek Shampoo   
49665       49661                                              Porto   
49666       49662                       Bacon Cheddar Pretzel Pieces   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   
49668       49664                     Lemon Cayenne Drinking Vinegar   
49669       49665      Super Dark Coconut Ash & Banana Chocolate Bar   
49670       49666                      Ginger Snaps Snacking Cookies   
49671       49667           Enchilada with Spanish Rice & Beans Meal   
49672       49668                     Apple Cinnamon Scented Candles   
49673       49669                                   K Cup Dark Blend   
49674       49670                                Beef Summer Sausage   
49675       49671                               Milk Chocolate Drops   
49676       49672                             Cafe Mocha K-Cup Packs   
49677       49673              Stone Baked Multi Grain Artisan Rolls   
49678       49674            Frozen Greek Yogurt Bars Chocolate Chip   
49679       49675                Cinnamon Dolce Keurig Brewed K Cups   
49680       49676                             Ultra Red Energy Drink   
49681       49677                    Thick & Chunky Sloppy Joe Sauce   
49682       49678                    Large Chicken & Cheese Taquitos   
49683       49679                            Famous Chocolate Wafers   
49684       49680                 All Natural Creamy Caesar Dressing   
49685       49681            Spaghetti with Meatballs and Sauce Meal   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
49658         5             13     4.0  
49659        98              7    10.7  
49660       128              3     7.1  
49661        83              4     8.3  
49662         2             16     3.9  
49663        17             13     3.1  
49664        22             11     6.7  
49665       134              5     8.2  
49666       107             19     3.6  
49667        57             14    10.2  
49668       100             21    13.7  
49669        45             19     6.9  
49670        61             19     5.2  
49671        38              1     6.6  
49672       101             17     5.6  
49673       100             21     4.7  
49674       106             12    19.2  
49675        45             19     3.0  
49676        26              7     6.5  
49677       129              1     5.6  
49678        37              1    11.1  
49679        26              7    14.0  
49680        64              7    14.5  
49681        59             15     8.9  
49682       129              1     3.4  
49683        61             19     6.0  
49684        89             13     4.9  
49685        38              1     6.9  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41     

In [12]:
# view the products dataframe columns
df_prods.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

In [13]:
# view the products dataframe dimensions
df_prods.shape

(49693, 5)

In [14]:
# products dataframe information
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49693 entries, 0 to 49692
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_id     49693 non-null  int64  
 1   product_name   49677 non-null  object 
 2   aisle_id       49693 non-null  int64  
 3   department_id  49693 non-null  int64  
 4   prices         49693 non-null  float64
dtypes: float64(1), int64(3), object(1)
memory usage: 1.9+ MB


In [15]:
# products dataframe describe
df_prods.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

### Answers to question 11

*What’s the maximum value of the “aisle_id” column?* 
134

*What’s the data type of the “department_id” column?* 
int64 (64-bit signed integers)

## 06. Examining orders.csv data

In [16]:
# Use the df.head function to view the first twenty rows of the orders dataframe
df.head(20)

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0    2539329        1             1          2                  8   
1    2398795        1             2          3                  7   
2     473747        1             3          3                 12   
3    2254736        1             4          4                  7   
4     431534        1             5          4                 15   
5    3367565        1             6          2                  7   
6     550135        1             7          1                  9   
7    3108588        1             8          1                 14   
8    2295261        1             9          1                 16   
9    2550362        1            10          4                  8   
10   1187899        1            11          4                  8   
11   2168274        2             1          2                 11   
12   1501582        2             2          5                 10   
13   1901567        2             3          1                 10   
14    738281        2             4          2                 10   
15   1673511        2             5          3                 11   
16   1199898        2             6          2                  9   
17   3194192        2             7          2                 12   
18    788338        2             8          1                 15   
19   1718559        2             9          2                  9   

    days_since_prior_order  
0                      NaN  
1                     15.0  
2                     21.0  
3                     29.0  
4                     28.0  
5                     19.0  
6                     20.0  
7                     14.0  
8                      0.0  
9                     30.0  
10                    14.0  
11                     NaN  
12                    10.0  
13                     3.0  
14                     8.0  
15                     8.0  
16                    13.0  
17                    14.0  
18                    27.0  
19                     8.0

In [17]:
# Use the df.tail function to view the last thirty five rows of the orders dataframe
df.tail(35)

order_id  user_id  order_number  order_dow  order_hour_of_day  \
3421048    726186   206208            30          0                 16   
3421049   2523609   206208            31          4                 11   
3421050    756221   206208            32          6                 18   
3421051   2254244   206208            33          6                 13   
3421052   2017995   206208            34          6                 15   
3421053   2789700   206208            35          3                 22   
3421054    844592   206208            36          6                 15   
3421055   1541132   206208            37          2                 12   
3421056   2808240   206208            38          0                 15   
3421057   3027766   206208            39          2                 14   
3421058   3356245   206208            40          5                  9   
3421059    442304   206208            41          2                 14   
3421060   2675140   206208            42          1                 19   
3421061    167903   206208            43          4                 14   
3421062   2393201   206208            44          6                 16   
3421063   3292671   206208            45          2                 11   
3421064   3059777   206208            46          1                 10   
3421065   2239861   206208            47          3                  4   
3421066   1285346   206208            48          1                 11   
3421067   1882108   206208            49          1                 22   
3421068    803273   206208            50          5                 11   
3421069   3154581   206209             1          3                 11   
3421070   1889163   206209             2          3                 17   
3421071   1542354   206209             3          5                 11   
3421072    688306   206209             4          1                 10   
3421073   2307371   206209             5          4                 15   
3421074   3186442   206209             6          0                 16   
3421075    550836   206209             7          2                 13   
3421076   2129269   206209             8          3                 17   
3421077   2558525   206209             9          4                 15   
3421078   2266710   206209            10          5                 18   
3421079   1854736   206209            11          4                 10   
3421080    626363   206209            12          1                 12   
3421081   2977660   206209            13          1                 12   
3421082    272231   206209            14          6                 14   

         days_since_prior_order  
3421048                     3.0  
3421049                     4.0  
3421050                     9.0  
3421051                     7.0  
3421052                     7.0  
3421053                     4.0  
3421054                    10.0  
3421055                     3.0  
3421056                    19.0  
3421057                     9.0  
3421058                    10.0  
3421059                    11.0  
3421060                     6.0  
3421061                     3.0  
3421062                     2.0  
3421063                     3.0  
3421064                    13.0  
3421065                     9.0  
3421066                     5.0  
3421067                     7.0  
3421068                     4.0  
3421069                     NaN  
3421070                     7.0  
3421071                    30.0  
3421072                    30.0  
3421073                     3.0  
3421074                     3.0  
3421075                     9.0  
3421076                    22.0  
3421077                    22.0  
3421078                    29.0  
3421079                    30.0  
3421080                    18.0  
3421081                     7.0  
3421082                    30.0

In [18]:
# orders dataframe dimensions
df.shape

(3421083, 6)

In [19]:
# orders dataframe information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3421083 entries, 0 to 3421082
Data columns (total 6 columns):
 #   Column                  Dtype  
---  ------                  -----  
 0   order_id                int64  
 1   user_id                 int64  
 2   order_number            int64  
 3   order_dow               int64  
 4   order_hour_of_day       int64  
 5   days_since_prior_order  float64
dtypes: float64(1), int64(5)
memory usage: 156.6 MB


In [20]:
# orders dataframe describe
df.describe()

order_id       user_id  order_number     order_dow  \
count  3.421083e+06  3.421083e+06  3.421083e+06  3.421083e+06   
mean   1.710542e+06  1.029782e+05  1.715486e+01  2.776219e+00   
std    9.875817e+05  5.953372e+04  1.773316e+01  2.046829e+00   
min    1.000000e+00  1.000000e+00  1.000000e+00  0.000000e+00   
25%    8.552715e+05  5.139400e+04  5.000000e+00  1.000000e+00   
50%    1.710542e+06  1.026890e+05  1.100000e+01  3.000000e+00   
75%    2.565812e+06  1.543850e+05  2.300000e+01  5.000000e+00   
max    3.421083e+06  2.062090e+05  1.000000e+02  6.000000e+00   

       order_hour_of_day  days_since_prior_order  
count       3.421083e+06            3.214874e+06  
mean        1.345202e+01            1.111484e+01  
std         4.226088e+00            9.206737e+00  
min         0.000000e+00            0.000000e+00  
25%         1.000000e+01            4.000000e+00  
50%         1.300000e+01            7.000000e+00  
75%         1.600000e+01            1.500000e+01  
max         2.300000e+01            3.000000e+01